### DISCUSSION TITLE
cache in redis

### DISCUSSION THREAD

**Discussion by Potisin** [long2ice/fastapi-cache#170] [Q&A] [answered]
hey.
Maybe I don’t fully understand, but why is the cache not stored in redis? everything is configured right and requests are cached but in the browser. Not in the redis (empty array).

**Comment by mjpieters** [comment]
You'll have to share how you have set up the extension.

- can you share how you call `FastAPICache.init()`?
- By 'cached in the browser', do you mean that the FastAPI responses have `Cache-Control` and `ETag` headers set?

We can't help without more detail, sorry!

**Comment by Potisin** [comment]
1. 
```
@app.on_event("startup")
async def startup():
    redis = aioredis.from_url("redis://localhost", encoding="utf8", decode_responses=True)
    FastAPICache.init(RedisBackend(redis), prefix="fastapi-cache")
```

2. Yup. This works even with a non-running Redis container. Perhaps I can somehow disable Cache-Control and ETag in the response?

**Reply by mjpieters** [reply]
Are you using the current main branch version or a specific release?

If you are using the main branch then you need to drop the `decode_responses` argument, [as documented](https://github.com/long2ice/fastapi-cache#redisbackend).

Otherwise please [enable logging in your application](https://github.com/long2ice/fastapi-cache/issues/119#issuecomment-1527900723) and share what output your see from the fastapi_cache loggers (if you include `%(name)s` in the logging format string it'll be very clear which log lines come from this project).

Another helpful piece of information would have been the response headers that your browser receives. If you are using the main branch version then there is a `X-FastAPI-Cache:` header in there that'll at least tell you if there was a cache hit or not.

**Reply by Potisin** [reply]
"Sorry, the problem was that I was running Redis in a Docker container. I tried running it locally, and caching works fine. 
However, I'm unable to establish a connection with the Redis container. Are there any instructions available for this? I've tried both localhost and the Docker host's IP address, and I even created a separate network for the application and the Docker container."

**Reply by mjpieters** [reply, marked as answer]
That'll depend on what operating system this is on, and how you are starting Redis. See [this Stack Overflow post](https://stackoverflow.com/questions/41371402/connecting-to-redis-running-in-docker-container-from-host-machine) for example, where Redis in the container is bound to localhost and not 0.0.0.0.

I'd search for "connect to docker redis from host" and research from there. Good luck!
